cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -87,185 +90,213 @@ public void addStateVariablesToCgmes() {
+            Bus b = getBusViewBus(vl, iidmNode);
             if (b != null) {
-                p.put(CgmesNames.ANGLE, fs(b.getAngle()));
-                p.put(CgmesNames.VOLTAGE, fs(b.getV()));
-                p.put(CgmesNames.TOPOLOGICAL_NODE, tn.getId(CgmesNames.TOPOLOGICAL_NODE));
-                voltages.add(p);
+                svVoltages.add(buildSvVoltage(tp, b.getV(), b.getAngle()));
+                processedTPNodes.add(tp);
             }
+        });
+        // FIXME(Luma) This processing should be common to bus-branch
+        // Buses not processed will be exported with the same values of the original CGMES data
+        originalTopologicalNodes.stream()
+            .filter(tn -> tn.get(CgmesNames.ANGLE) != null && tn.get(CgmesNames.VOLTAGE) != null)
+            .filter(tn -> !processedTPNodes.contains(tn.getId(CgmesNames.TOPOLOGICAL_NODE)))
+            .filter(tn -> !boundaryNodesFromDanglingLines.values().contains(tn.getId(CgmesNames.TOPOLOGICAL_NODE)))
+            .forEach(pb -> {
+                String tnId = pb.getId(CgmesNames.TOPOLOGICAL_NODE);
+                double v = pb.asDouble(CgmesNames.VOLTAGE);
+                double angle = pb.asDouble(CgmesNames.ANGLE);
+                svVoltages.add(buildSvVoltage(tnId, v, angle));
+            });
+    }
+
+    private static Bus getBusViewBus(VoltageLevel vl, int iidmNode) {
+        // We use IIDM API to locate node-breaker buses
+        VoltageLevel.NodeBreakerView topo = vl.getNodeBreakerView();
+        if (!topo.hasAttachedEquipment(iidmNode)) {
+            LOG.error("IIDM node {} not valid", iidmNode);
+            return null;
+        }
+        // If there is no Terminal at this IIDM node,
+        // then find from it the first connected node with a Terminal
+        Terminal t = topo.getOptionalTerminal(iidmNode)
+                .orElseGet(() -> Networks.getEquivalentTerminal(vl, iidmNode));
+        if (t == null) {
+            LOG.error("Can't find a Terminal for IIDM node {}", iidmNode);
+            return null;
         }
-        cgmes.add(originalSVcontext, "SvVoltage", voltages);
+        return t.getBusView().getBus();
     }
 
     private void addVoltagesForBoundaryNodes() {
-        PropertyBags voltages = new PropertyBags();
-        // add voltages for TpNodes existing in the Model's boundaries
-        boundaryNodesFromDanglingLines.forEach((line, bnode) -> {
-            PropertyBag p = new PropertyBag(SV_VOLTAGE_PROPERTIES);
+        // Add voltages for TP nodes located at boundaries
+        PropertyBags svVoltages = new PropertyBags();
+        boundaryNodesFromDanglingLines.forEach((line, tpNodeId) -> {
             DanglingLine dl = network.getDanglingLine(line);
             Bus b = dl.getTerminal().getBusBreakerView().getBus();
+            double v;
+            double angle;
             if (b != null) {
-                // calculate complex voltage value: abs for VOLTAGE, degrees for ANGLE
```

Comment:
Indeed we have 2 solutions :
- Recompute the angle and the voltage at boundary buses ;
- Or use the values computed by the loadflow engine with the use of the extension ````MergedXnode```` or properties.
I think that we have to support both options.